components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/editor/js/design-view/cron-generator/cron-generator.js

```diff
@@ -0,0 +1,771 @@
+            if (dayOfWeek.search(/[^\w-,*\/#]/) !== -1) {
+                return false;
+            }
+            var list = dayOfWeek.split(",");
+            return list.every(function (dayOfWeek){
+                if(dayOfWeek.includes('/') && dayOfMonth === '?') {
+                    var startingDayOfWeekOptionArr = dayOfWeek.split('/');
+
+                    var isValidElements = (isValidateNumericValue([startingDayOfWeekOptionArr[0]], 1, 7) &&
+                                                        isValidateNumericValue([startingDayOfWeekOptionArr[1]], 1, 7));
+                    var isValidFirstElem = (startingDayOfWeekOptionArr[0] === '*' &&
+                                                        isValidateNumericValue([startingDayOfWeekOptionArr[1]], 1, 7));
+                    return isValidElements || isValidFirstElem;
+
+                } else if(dayOfWeek.includes('-') && dayOfMonth === '?') {
+                    var dayOfWeekRangeArr = dayOfWeek.split('-');
+                    var validWeekRange = parseInt(dayOfWeekRangeArr[0]) < parseInt(dayOfWeekRangeArr[1]);
+                    var validWeekStrRange = WEEK_ARRRAY.indexOf(dayOfWeekRangeArr[0]) < WEEK_ARRRAY.indexOf(dayOfWeekRangeArr[1]);
+                    return !isNaN(parseInt(dayOfWeekRangeArr[0])) && !isNaN(parseInt(dayOfWeekRangeArr[1])) ?
+                        isValidateNumericValue(dayOfWeekRangeArr, 1, 7) && validWeekRange :
+                        isValidateStringValue(dayOfWeekRangeArr, WEEK_ARRRAY) && validWeekStrRange;
+
+                } else if(dayOfWeek.includes('#') && dayOfMonth === '?') {
+                    var weekdayOfMonthArr = dayOfWeek.split('#');
+                        return (isValidateStringValue([weekdayOfMonthArr[0]], WEEK_ARRRAY) &&
+                                isValidateNumericValue([weekdayOfMonthArr[1]], 1, 5)) ||
+                                 (isValidateNumericValue([weekdayOfMonthArr[0]], 1, 7) &&
+                                 isValidateNumericValue([weekdayOfMonthArr[1]], 1, 5));
+
+                } else if(typeof dayOfWeek === 'string' && dayOfMonth === '?') {
+                    var firstIndexValue = dayOfWeek.charAt(0);
+                    var secondIndexValue = dayOfWeek.charAt(1);
+                    if(dayOfWeek.length === 1){
+                        return isValidateNumericValue([dayOfWeek], 1, 7)
+                    } else {
+                        return (!isNaN(parseInt(firstIndexValue)) && !isNaN(parseInt(secondIndexValue))) ?
+                                isValidateNumericValue([dayOfWeek], 1, 7) : isValidateStringValue([dayOfWeek], WEEK_ARRRAY);
+                    }
+                } else {
+                    return false;
+                }
+            });
+        };
+
+        var isValidCronExpression = function(cronExpression) {
+            if(!/\s/g.test(cronExpression)) {
+                return false;
+            }
+            var cronArray = cronExpression.split(" ");
```

Comment:
Did you check the possibility where user copy paste a cron experession without spaces ??

Reply:
yes

Reply:
so wt happens when user copy a valid cron expresion without spaces ?

Reply:
It will says as a Invalid cron expression.

Reply:
That is wrong ne because it is not a invalid cron expression. we don't have a rule saying cron expression should have each section space delimited right ? @lasanthafdo what can we do here ?

Reply:
@lasanthafdo wdyt ?